File: leetcode/228_SummaryRanges/228_Summary_Ranges.py
class Solution:
    def output(self, start, prev):
        if start == prev:
            return str(start)
        else:
            return str(start) + "->" + str(prev)
        
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        if len(nums) == 0:
            return []
        start = nums[0]
        prev = nums[0]
        res = []
        for i in range(1, len(nums)):
            if nums[i] == prev + 1:
                prev = nums[i]
            else:
                res.append(self.output(start, prev))
                start = nums[i]
                prev = nums[i]
        if prev == nums[-1]:
            res.append(self.output(start, prev))
        return res
        

File: leetcode/228_SummaryRanges/228_Summary_Ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> res = new ArrayList<>();
        if (nums == null) return null;
        if (nums.length == 0) return res;
        
        int start = nums[0];
        int startIndex = 0;
        for (int i = 1; i < nums.length; i++) {
            if ((long)nums[i] - (long)nums[i - 1] > 1) {
                if (startIndex == i - 1) res.add(start + "");
                else res.add(start + "->" + nums[i - 1]);
                start = nums[i];
                startIndex = i;
            }
        }
        if (startIndex == nums.length - 1) res.add(start + "");
        else res.add(start + "->" + nums[nums.length - 1]);
        return res;
    }
}
